src/world.rs

```diff
@@ -47,15 +48,19 @@ pub struct World {
     archetypes: ArchetypeSet,
     /// Maps statically-typed bundle types to archetypes
     bundle_to_archetype: TypeIdMap<u32>,
+    id: usize,
 }
 
 impl World {
     /// Create an empty world
     pub fn new() -> Self {
+        static NEXT_ID: AtomicUsize = AtomicUsize::new(0);
+
         Self {
             entities: Entities::default(),
             archetypes: ArchetypeSet::new(),
             bundle_to_archetype: HashMap::default(),
+            id: NEXT_ID.fetch_add(1, Ordering::Relaxed),
```

Comment:
This should probably guard against overflow like
```rust
NEXT_ID.fetch_update(Ordering::Relaxed, Ordering::Relaxed, |next_id| next_id.checked_add(1)).unwrap()
```

Reply:
If we're relying on it for soundness, yeah. Alternatively, if we use an `AtomicU64` then it's probably reasonable to assume that overflow won't happen.